x-pack/test/cases_api_integration/common/fixtures/plugins/security_solution/server/plugin.ts

```diff
@@ -21,9 +30,58 @@ export interface FixtureStartDeps {
 }
 
 export class FixturePlugin implements Plugin<void, void, FixtureSetupDeps, FixtureStartDeps> {
+  private readonly logger: Logger;
+
+  private securityStart?: SecurityPluginStart;
+  constructor(initContext: PluginInitializerContext) {
+    this.logger = initContext.logger.get();
+  }
+
   public setup(core: CoreSetup<FixtureStartDeps>, deps: FixtureSetupDeps) {
+    const router = core.http.createRouter();
+    this.registerRoutes(router);
+
     const { features } = deps;
+    this.registerFeatures(features);
+  }
+
+  public start(core: CoreStart, plugins: FixtureStartDeps) {
+    this.securityStart = plugins.security;
+  }
+
+  public stop() {}
+
+  private registerRoutes(router: IRouter<RequestHandlerContext>) {
```

Comment:
I think instead of creating our own router and maintaining it we can use the`/internal/security/user_profiles/_bulk_get` endpoint provided by the security plugin.